cmd/talosctl/cmd/mgmt/cluster/create.go

```diff
@@ -746,9 +750,38 @@ func create(ctx context.Context, flags *pflag.FlagSet) error {
 		extraKernelArgs = procfs.NewCmdline(extraBootKernelArgs)
 	}
 
+	if withSiderolinkAgent {
+		if extraKernelArgs == nil {
+			extraKernelArgs = procfs.NewCmdline("")
+		}
+
+		if extraKernelArgs.Get("siderolink.api") != nil || extraKernelArgs.Get("talos.events.sink") != nil || extraKernelArgs.Get("talos.logging.kernel") != nil {
+			return errors.New("siderolink kernel arguments are already set, cannot run with --with-siderolink")
+		}
+
+		wgPort := rand.IntN(20000) + 40000
```

Comment:
instead of this, do something like https://github.com/siderolabs/talos/pull/8476/files#diff-6ed6be17858f4d4a230f75c46a3ade864e225e78d87760e13788db4bb58fe534R261-R284

Reply:
Done. We now generate ports in `getDynamicPorts` which uses `getDynamicPort` and checks that generated ports do not overlap.